internal/pkg/override/patch.go

```diff
@@ -0,0 +1,253 @@
+				Tag:   "!!str",
+				Value: pointer[0],
+			})
+			node.Content = append(node.Content, &y.Value)
+			return errStopFollowingPointer
+		case yaml.SequenceNode:
+			if pointer[0] == "-" || pointer[0] == "" {
+				// add to end of sequence
+				node.Content = append(node.Content, &y.Value)
+				return errStopFollowingPointer
+			}
+
+			idx, err := strconv.Atoi(pointer[0])
+			switch {
+			case err != nil:
+				return fmt.Errorf("expected index in sequence, got %q", pointer[0])
+			case idx < 0 || idx > len(node.Content)-1:
+				return fmt.Errorf("invalid index %d for sequence of length %d", idx, len(node.Content))
+			}
+
+			// add node at idx
+			node.Content = append(node.Content[:idx], append([]*yaml.Node{&y.Value}, node.Content[idx:]...)...)
+			return errStopFollowingPointer
+		}
+
+		return nil
+	})
+}
+
+func (y *yamlPatch) doRemove(root *yaml.Node) error {
+	return followJSONPointer(root, y.Path, func(node *yaml.Node, pointer []string) error {
+		if len(pointer) != 1 {
+			return nil
+		}
+
+		switch node.Kind {
+		case yaml.MappingNode:
+			for i := 0; i < len(node.Content); i += 2 {
+				if node.Content[i].Value == pointer[0] {
+					node.Content = append(node.Content[:i], node.Content[i+2:]...)
+					return errStopFollowingPointer
+				}
+			}
+		case yaml.SequenceNode:
+			idx, err := strconv.Atoi(pointer[0])
+			switch {
+			case err != nil:
+				return fmt.Errorf("expected index in sequence, got %q", pointer[0])
+			case idx < 0 || idx > len(node.Content)-1:
+				return fmt.Errorf("invalid index %d for sequence of length %d", idx, len(node.Content))
+			}
+
+			node.Content = append(node.Content[:idx], node.Content[idx+1:]...)
+			return errStopFollowingPointer
+		}
+
+		return nil
+	})
+}
+
+func (y *yamlPatch) applyReplace(root *yaml.Node) error {
+	return followJSONPointer(root, y.Path, func(node *yaml.Node, pointer []string) error {
+		if len(pointer) > 0 {
+			return nil
+		}
+		return y.encodeAndStop(node)
+	})
+}
+
+func (y *yamlPatch) encodeAndStop(node *yaml.Node) error {
+	if err := node.Encode(y.Value); err != nil {
+		return err
+	}
+	return errStopFollowingPointer
+}
+
+var errStopFollowingPointer = errors.New("stop following pointer")
+
+func followJSONPointer(root *yaml.Node, pointer string, visit func(node *yaml.Node, pointer []string) error) error {
+	split := strings.Split(pointer, "/")
+	for i := range split {
+		split[i] = strings.ReplaceAll(split[i], "~0", "~")
+		split[i] = strings.ReplaceAll(split[i], "~1", "/")
+	}
+
+	return followJSONPointerHelper(root, split, visit)
+}
+
+func followJSONPointerHelper(node *yaml.Node, pointer []string, visit func(node *yaml.Node, pointer []string) error) error {
+	if err := visit(node, pointer); err != nil {
+		if errors.Is(err, errStopFollowingPointer) {
+			return nil
+		}
+		return err
+	} else if len(pointer) == 0 {
+		return nil
+	}
+
+	switch node.Kind {
+	case yaml.DocumentNode:
+		if len(node.Content) != 1 {
+			return fmt.Errorf("don't support multi-doc yaml")
```

Comment:
you can [embed two yaml documents in a single file](https://yaml.org/spec/1.2.2/#22-structures). since multi document files don't exist in JSON (and therefore JSON Pointer syntax), there is no way to target a specific document with a JSON Pointer. So we just assume the target is the first document, and error if there are multiple documents in the file.

Reply:
update: https://pkg.go.dev/gopkg.in/yaml.v3#Unmarshal 🤦 
> Unmarshal decodes the first document

Since we only call unmarshal once, I think it's ok if we just ignore this case.